bgbb/core.py

```diff
@@ -113,3 +119,27 @@ def cond_exp_rets_till_nb(self, t, frequency, recency, n, params):
         p1 = 1 / exp(self._loglikelihood(params, xa, txa, na))
         ret = cond_exp_rets_till_p2345(t, xa, txa, na, params, p1)
         return ret
+
+    @classmethod
+    def _log_expec_p_th(cls, a: float, b: float, g: float, d: float, f, r, n):
+        l_ap1 = cls._loglikelihood([a + 1, b, g, d], f, r, n)
+        l_gp1 = cls._loglikelihood([a, b, g + 1, d], f, r, n)
+        ll = cls._loglikelihood([a, b, g, d], f, r, n)
+
+        l_a_frac = np.log(a) - np.log(a + b)
+        l_g_frac = np.log(g) - np.log(g + d)
+
+        return (l_a_frac + l_ap1 - ll), (l_g_frac + l_gp1 - ll)
+
+    @classmethod
+    def _expec_p_th(cls, a, b, g, d, f, r, n):
+        lp_exp, lth_exp = cls._log_expec_p_th(a, b, g, d, f, r, n)
+        return np.exp(lp_exp), np.exp(lth_exp)
+
+    def expec_p_th(self, f, r, n) -> Tuple[Sequence[Prob], Sequence[Prob]]:
```

Comment:
As a public method, it would be nice to have a readable name like `get_latent_variables`.

Reply:
changed to `latent_variable_mean`